maven-plugin/src/main/java/org/openmrs/maven/plugins/Setup.java

```diff
@@ -417,26 +371,112 @@ private void setDebugPort(Server server) {
-        DistroProperties distroProperties;
-        if(DistroHelper.isRefapp2_3_1orLower(server.getDistroArtifactId(), server.getVersion())){
-            distroProperties = new DistroProperties(server.getVersion());
-        } else {
-            distroProperties = distroHelper.downloadDistroProperties(serverPath, server);
+    private void setServerVersionsFromDistroProperties(Server server, DistroProperties distroProperties) throws MojoExecutionException {
+        if(server.getPlatformVersion() == null){
+            server.setPlatformVersion(distroProperties.getPlatformVersion(distroHelper, server.getServerDirectory()));
+        }
+        if(server.getVersion() == null){
+            server.setVersion(distroProperties.getVersion());
         }
-        moduleInstaller.installCoreModules(server, isCreatePlatform, distroProperties, distroHelper);
-        installOWAs(server, distroProperties);
-        return distroProperties;
     }
 
-    private void configureVersion(Server server, DistroProperties distroProperties) throws MojoExecutionException {
-        if (distroProperties != null) {
-            if(server.getPlatformVersion() == null){
-                server.setPlatformVersion(distroProperties.getPlatformVersion(distroHelper, server.getServerDirectory()));
+    private void setupDatabase(Server server, DistroProperties distroProperties) throws MojoExecutionException {
+        if (server.getDbDriver() == null) {
+            boolean isH2Supported = true;
+            if (distroProperties != null) {
+                isH2Supported = distroProperties.isH2Supported();
+            }
+            wizard.promptForDb(server, dockerHelper, isH2Supported, dbDriver, dockerHost);
+        }
+        if (server.getDbDriver() != null) {
+            setupDatabaseForServer(server);
+        }
+    }
+
+    private void setupDatabaseForServer(Server server) throws MojoExecutionException {
+        if(server.getDbName() == null){
+            server.setDbName(determineDbName(server.getDbUri(), server.getServerId()));
+        }
+        if (server.isMySqlDb()){
+            boolean mysqlDbCreated = connectMySqlDatabase(server);
+            if(!mysqlDbCreated){
+                throw new IllegalStateException("Failed to connect to the specified database " + server.getDbUri());
+            }
+
+            if (hasDbTables(server)) {
+                if (dbReset == null) {
+                    dbReset = !wizard.promptYesNo("Would you like to setup the server using existing data (if not, all data will be lost)?");
+                }
+
+                if (dbReset) {
+                    wipeDatabase(server);
+                } else {
+                    server.setParam("create_tables", "false");
+                }
+            }
+
+            if (dbReset == null){
+                dbReset = true;
+            }
+
+
+            if(!"null".equals(dbSql) && dbReset){
```

Comment:
Not that someone shouldn't refactor this at some point

Reply:
The blocks for mysql and postgres are identical except for a couple functions. Easy refactor in a functional language...